nodeclient/core_models.go

```diff
@@ -93,113 +108,82 @@ type (
+		TxState string `json:"txState,omitempty"`
+		// BlockStateReason if applicable indicates the error that occurred during the block processing.
+		BlockStateReason string `json:"blockStateReason,omitempty"`
+		// TxStateReason if applicable indicates the error that occurred during the transaction processing.
+		TxStateReason string `json:"txStateReason,omitempty"`
+		// ReissuePayload whether the block should be reissued.
+		ReissuePayload *bool `json:"reissuePayload,omitempty"`
 	}
 
 	// OutputMetadataResponse defines the response of a GET outputs metadata REST API call.
 	OutputMetadataResponse struct {
-		// The hex encoded ID of the block.
+		// BlockID is the block ID that contains the output.
 		BlockID string `json:"blockId"`
-		// The hex encoded transaction id from which this output originated.
+		// TransactionID is the transaction ID that creates the output.
 		TransactionID string `json:"transactionId"`
-		// The index of the output.
+		// OutputIndex is the index of the output.
 		OutputIndex uint16 `json:"outputIndex"`
-		// Whether this output is spent.
-		Spent bool `json:"isSpent"`
-		// The milestone index at which this output was spent.
-		MilestoneIndexSpent iotago.SlotIndex `json:"milestoneIndexSpent,omitempty"`
-		// The milestone timestamp this output was spent.
-		MilestoneTimestampSpent uint32 `json:"milestoneTimestampSpent,omitempty"`
-		// The transaction this output was spent with.
+		// IsSpent indicates whether the output is spent or not.
+		IsSpent bool `json:"isSpent"`
+		// CommitmentIDSpent is the commitment ID that includes the spent output.
+		CommitmentIDSpent string `json:"commitmentIdSpent,omitempty"`
+		// TransactionIDSpent is the transaction ID that spends the output.
 		TransactionIDSpent string `json:"transactionIdSpent,omitempty"`
-		// The milestone index at which this output was booked into the ledger.
-		MilestoneIndexBooked iotago.SlotIndex `json:"milestoneIndexBooked"`
-		// The milestone timestamp this output was booked in the ledger.
-		MilestoneTimestampBooked uint32 `json:"milestoneTimestampBooked"`
-		// The ledger index at which this output was available at.
-		LedgerIndex iotago.SlotIndex `json:"ledgerIndex"`
-	}
-
-	// OutputResponse defines the response of a GET outputs REST API call.
-	OutputResponse struct {
-		Metadata *OutputMetadataResponse `json:"metadata"`
-		// The output in its serialized form.
-		RawOutput *json.RawMessage `json:"output"`
-	}
-
-	// TreasuryResponse defines the response of a GET treasury REST API call.
-	TreasuryResponse struct {
-		MilestoneID string `json:"milestoneId"`
-		Amount      string `json:"amount"`
-	}
-
-	// ReceiptsResponse defines the response for receipts GET related REST API calls.
-	ReceiptsResponse struct {
-		Receipts []*ReceiptTuple `json:"receipts"`
+		// IncludedCommitmentID is the commitment ID that includes the output.
+		IncludedCommitmentID string `json:"includedCommitmentId,omitempty"`
```

Comment:
What happens if an output is not committed yet and a client requests it? Is EmptyCommittment returned?